--- a/chrome/browser/ui/views/frame/browser_frame_view_win.cc
+++ b/chrome/browser/ui/views/frame/browser_frame_view_win.cc
@@ -624,11 +624,12 @@ int BrowserFrameViewWin::TopAreaHeight(b
   }
 
   const bool maximized = IsMaximized() && !restored;
-  int top = FrameTopBorderThickness(restored);
+  int top = WindowTopY();
   if (IsWebUITabStrip()) {
     // Caption bar is default Windows size in maximized mode but full size when
     // windows are tiled in tablet mode (baesd on behavior of first-party
     // Windows applications).
+    top = FrameTopBorderThickness(restored);
     top += maximized ? TitlebarMaximizedVisualHeight()
                      : caption_button_container_->GetPreferredSize().height();
     return top;
