--- a/chrome/app/theme/chromium/BRANDING
+++ b/chrome/app/theme/chromium/BRANDING
@@ -1,10 +1,10 @@
-COMPANY_FULLNAME=The Helium Authors
-COMPANY_SHORTNAME=The Helium Authors
-PRODUCT_FULLNAME=Helium
-PRODUCT_SHORTNAME=Helium
-PRODUCT_INSTALLER_FULLNAME=Helium Installer
-PRODUCT_INSTALLER_SHORTNAME=Helium Installer
-COPYRIGHT=Copyright @LASTCHANGE_YEAR@ The Helium Authors. All rights reserved.
-MAC_BUNDLE_ID=net.imput.helium
+COMPANY_FULLNAME=The Neon Authors
+COMPANY_SHORTNAME=The Neon Authors
+PRODUCT_FULLNAME=Neon
+PRODUCT_SHORTNAME=Neon
+PRODUCT_INSTALLER_FULLNAME=Neon Installer
+PRODUCT_INSTALLER_SHORTNAME=Neon Installer
+COPYRIGHT=Copyright @LASTCHANGE_YEAR@ The Neon Authors. All rights reserved.
+MAC_BUNDLE_ID=com.getxuxes.neon
 MAC_CREATOR_CODE=Cr24
-MAC_TEAM_ID=S4Q33XPHB4
+MAC_TEAM_ID=
--- a/chrome/app/chrome_crash_reporter_client.cc
+++ b/chrome/app/chrome_crash_reporter_client.cc
@@ -135,7 +135,7 @@ void ChromeCrashReporterClient::GetProductInfo(ProductInfo* product_info) {
   CHECK(product_info);

-  const char* product_name = "Helium";
+  const char* product_name = "Neon";
   *product_info =
       ProductInfo(product_name, version_info::GetHeliumVersionNumber(),
                   chrome::GetChannelName(chrome::WithExtendedStable(true)));
 }
