--- a/chrome/browser/ui/views/webid/fedcm_account_selection_view_desktop.cc
+++ b/chrome/browser/ui/views/webid/fedcm_account_selection_view_desktop.cc
@@ -1294,6 +1294,9 @@ void FedCmAccountSelectionView::ShouldSh
   }
 
   if (dialog_type_ == DialogType::BUBBLE) {
+    should_show = false;
+    return;
+
     // Hide the bubble dialog if it can't fit.
     if (!CanFitInWebContents()) {
       should_show = false;
