--- a/chrome/browser/ui/ui_features.cc
+++ b/chrome/browser/ui/ui_features.cc
@@ -129,7 +129,7 @@ BASE_FEATURE(kExtensionsPinnedByDefault,
 #if BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC)
 // Shows an infobar on PDFs offering to become the default PDF viewer if Chrome
 // isn't the default already.
-BASE_FEATURE(kPdfInfoBar, base::FEATURE_ENABLED_BY_DEFAULT);
+BASE_FEATURE(kPdfInfoBar, base::FEATURE_DISABLED_BY_DEFAULT);
 
 BASE_FEATURE(kSeparateDefaultAndPinPrompt,
 #if BUILDFLAG(IS_WIN)
--- a/chrome/browser/startup/startup_features.cc
+++ b/chrome/browser/startup/startup_features.cc
@@ -39,18 +39,7 @@ bool IsForegroundLaunchEnabled() {
 }
 
 bool IsForegroundLaunchInfoBarEnabled() {
-  if (!IsForegroundLaunchEnabled()) {
-    return false;
-  }
-  const std::optional<bool> infobar_override =
-      base::FeatureList::GetStateIfOverridden(kLaunchOnStartupInfoBar);
-  if (!infobar_override.has_value()) {
-    // If the infobar feature flag is not available / configured, use the base
-    // feature flag.
-    return true;
-  }
-
-  return *infobar_override;
+  return false;
 }
 
 LaunchOnStartupDefaultPreference GetLaunchOnStartupDefaultPreference() {
