--- a/chrome/browser/helium_flag_choices.h
+++ b/chrome/browser/helium_flag_choices.h
@@ -27,6 +27,7 @@ namespace helium {
   };
 
   constexpr const char kDisableEchCommandLine[] = "disable-ech";
+  constexpr const char kCanvasNoiseCommandLine[] = "fingerprinting-canvas-noise";
 
 }  // namespace helium
 
--- a/chrome/browser/helium_flag_entries.h
+++ b/chrome/browser/helium_flag_entries.h
@@ -16,4 +16,8 @@
      "Disables TLS Encrypted Client Hello. Not recommended unless you live in an area with heavy Internet"
      " censorship and ECH prevents websites from loading. Helium flag.",
      kOsAll, SINGLE_VALUE_TYPE(helium::kDisableEchCommandLine)},
+    {helium::kCanvasNoiseCommandLine,
+     "Canvas fingerprint deception",
+     "Enable noising pixels when the contents of a canvas are read back by a script. Bound to a browser session. Helium flag, Chromium feature.",
+     kOsAll, SINGLE_VALUE_TYPE(helium::kCanvasNoiseCommandLine)},
 #endif  /* CHROME_BROWSER_HELIUM_FLAG_ENTRIES_H_ */
