--- a/chrome/browser/ui/bookmarks/bookmark_bar_controller.cc
+++ b/chrome/browser/ui/bookmarks/bookmark_bar_controller.cc
@@ -144,10 +144,6 @@
 }
 
 bool BookmarkBarController::ShouldShowBookmarkBar() const {
-  if (base::CommandLine::ForCurrentProcess()->GetSwitchValueASCII("bookmark-bar-ntp") == "never") {
-    return false;
-  }
-
   Profile* profile = browser_->GetProfile();
   if (profile->IsGuestSession()) {
     return false;
@@ -200,6 +196,10 @@
     return false;
   }
 
+  if (base::CommandLine::ForCurrentProcess()->GetSwitchValueASCII("bookmark-bar-ntp") == "never") {
+    return false;
+  }
+
   return std::any_of(
       tabs.begin(), tabs.end(), [](const tabs::TabInterface* tab) {
         return tab->GetContents() && IsShowingNTP(tab->GetContents());
