--- a/chrome/browser/ui/accelerator_table.cc
+++ b/chrome/browser/ui/accelerator_table.cc
@@ -75,6 +75,8 @@ const AcceleratorMapping kAcceleratorMap[] = {
     {ui::VKEY_R, ui::EF_SHIFT_DOWN | ui::EF_PLATFORM_ACCELERATOR,
      IDC_RELOAD_BYPASSING_CACHE},
     {ui::VKEY_S, ui::EF_PLATFORM_ACCELERATOR, IDC_SAVE_PAGE},
+    {ui::VKEY_S, ui::EF_SHIFT_DOWN | ui::EF_PLATFORM_ACCELERATOR,
+     IDC_SHARING_HUB_SCREENSHOT},
     {ui::VKEY_9, ui::EF_PLATFORM_ACCELERATOR, IDC_SELECT_LAST_TAB},
     {ui::VKEY_NUMPAD9, ui::EF_PLATFORM_ACCELERATOR, IDC_SELECT_LAST_TAB},
 #if BUILDFLAG(IS_LINUX)
--- a/chrome/browser/global_keyboard_shortcuts_mac.mm
+++ b/chrome/browser/global_keyboard_shortcuts_mac.mm
@@ -153,6 +153,7 @@ const std::vector<KeyboardShortcutData>& GetShortcutsNotPresentInMainMenu() {
       {true,  false, false, true,  kVK_UpArrow,           IDC_FOCUS_PREVIOUS_PANE},
       {true,  true,  false, true,  kVK_ANSI_A,            IDC_FOCUS_INACTIVE_POPUP_FOR_ACCESSIBILITY},
       {true,  false, false, true,  kVK_ANSI_R,            IDC_SHOW_READING_MODE_KEYBOARD},
+      {true,  true,  false, false, kVK_ANSI_S,            IDC_SHARING_HUB_SCREENSHOT},
     });
     // clang-format on

--- a/chrome/browser/ui/cocoa/accelerators_cocoa.mm
+++ b/chrome/browser/ui/cocoa/accelerators_cocoa.mm
@@ -48,6 +48,8 @@ const struct AcceleratorMapping {
     {IDC_PRINT, ui::EF_COMMAND_DOWN, ui::VKEY_P},
     {IDC_RESTORE_TAB, ui::EF_COMMAND_DOWN | ui::EF_SHIFT_DOWN, ui::VKEY_T},
     {IDC_SAVE_PAGE, ui::EF_COMMAND_DOWN, ui::VKEY_S},
+    {IDC_SHARING_HUB_SCREENSHOT, ui::EF_COMMAND_DOWN | ui::EF_SHIFT_DOWN,
+     ui::VKEY_S},
     {IDC_SHOW_BOOKMARK_BAR, ui::EF_COMMAND_DOWN | ui::EF_SHIFT_DOWN,
      ui::VKEY_B},
     {IDC_SHOW_BOOKMARK_MANAGER, ui::EF_COMMAND_DOWN | ui::EF_ALT_DOWN,
--- a/chrome/app/chrome_dll.rc
+++ b/chrome/app/chrome_dll.rc
@@ -85,6 +85,7 @@ BEGIN
     VK_F5,          IDC_RELOAD_BYPASSING_CACHE, VIRTKEY, SHIFT
     "T",            IDC_RESTORE_TAB,            VIRTKEY, CONTROL, SHIFT
     "S",            IDC_SAVE_PAGE,              VIRTKEY, CONTROL
+    "S",            IDC_SHARING_HUB_SCREENSHOT, VIRTKEY, CONTROL, SHIFT
     "9",            IDC_SELECT_LAST_TAB,        VIRTKEY, CONTROL
     VK_NUMPAD9,     IDC_SELECT_LAST_TAB,        VIRTKEY, CONTROL
     VK_NEXT,        IDC_MOVE_TAB_NEXT,          VIRTKEY, CONTROL, SHIFT
