--- a/chrome/browser/ui/browser_command_controller.cc
+++ b/chrome/browser/ui/browser_command_controller.cc
@@ -453,9 +453,10 @@ bool BrowserCommandController::IsReserve
          command_id == IDC_NEW_WINDOW || command_id == IDC_RESTORE_TAB ||
          command_id == IDC_SELECT_NEXT_TAB ||
          command_id == IDC_SELECT_PREVIOUS_TAB || command_id == IDC_EXIT ||
          command_id == IDC_COPY_OR_INSPECT_SHORTCUT || command_id == IDC_DEV_TOOLS ||
-         command_id == IDC_DEV_TOOLS_TOGGLE || command_id == IDC_TAB_SEARCH;
+         command_id == IDC_DEV_TOOLS_TOGGLE || command_id == IDC_TAB_SEARCH ||
+         command_id == IDC_DUPLICATE_TAB;
 }

 void BrowserCommandController::TabStateChanged() {
--- a/chrome/browser/ui/cocoa/accelerators_cocoa.mm
+++ b/chrome/browser/ui/cocoa/accelerators_cocoa.mm
@@ -48,6 +48,8 @@ const struct AcceleratorMapping {
     {IDC_PASTE, ui::EF_COMMAND_DOWN, ui::VKEY_V},
     {IDC_PRINT, ui::EF_COMMAND_DOWN, ui::VKEY_P},
     {IDC_RESTORE_TAB, ui::EF_COMMAND_DOWN | ui::EF_SHIFT_DOWN, ui::VKEY_T},
+    {IDC_RESTORE_TAB, ui::EF_COMMAND_DOWN, ui::VKEY_Z},
+    {IDC_DUPLICATE_TAB, ui::EF_COMMAND_DOWN, ui::VKEY_D},
     {IDC_CTRL_S_SHORTCUT, ui::EF_COMMAND_DOWN, ui::VKEY_S},
     {IDC_SHOW_BOOKMARK_BAR, ui::EF_COMMAND_DOWN | ui::EF_SHIFT_DOWN,
      ui::VKEY_B},
--- a/chrome/browser/global_keyboard_shortcuts_mac.mm
+++ b/chrome/browser/global_keyboard_shortcuts_mac.mm
@@ -143,6 +143,10 @@ const std::vector<KeyboardShortcutData>&
       {true,  false, false, false, kVK_ANSI_S,            IDC_CTRL_S_SHORTCUT},

       {true,  true,  false, false, kVK_ANSI_M,            IDC_SHOW_AVATAR_MENU},
+
+      {true,  false, false, false, kVK_ANSI_D,            IDC_DUPLICATE_TAB},
+      {true,  false, false, false, kVK_ANSI_Z,            IDC_RESTORE_TAB},
+
       {true,  false, false, true,  kVK_ANSI_L,            IDC_SHOW_DOWNLOADS},
       {true,  true,  false, false, kVK_ANSI_I,            IDC_DEV_TOOLS},
       {true,  true,  false, false, kVK_ANSI_E,            IDC_DEV_TOOLS_INSPECT},
--- a/chrome/browser/ui/accelerator_table.cc
+++ b/chrome/browser/ui/accelerator_table.cc
@@ -70,6 +70,8 @@ const AcceleratorMapping kAcceleratorMap
     {ui::VKEY_R, ui::EF_PLATFORM_ACCELERATOR, IDC_RELOAD},
     {ui::VKEY_R, ui::EF_SHIFT_DOWN | ui::EF_PLATFORM_ACCELERATOR,
      IDC_RELOAD_BYPASSING_CACHE},
+    {ui::VKEY_D, ui::EF_PLATFORM_ACCELERATOR, IDC_DUPLICATE_TAB},
+    {ui::VKEY_Z, ui::EF_PLATFORM_ACCELERATOR, IDC_RESTORE_TAB},
     {ui::VKEY_S, ui::EF_PLATFORM_ACCELERATOR, IDC_CTRL_S_SHORTCUT},
     {ui::VKEY_9, ui::EF_PLATFORM_ACCELERATOR, IDC_SELECT_LAST_TAB},
     {ui::VKEY_NUMPAD9, ui::EF_PLATFORM_ACCELERATOR, IDC_SELECT_LAST_TAB},
--- a/chrome/browser/ui/views/frame/browser_native_widget_mac.mm
+++ b/chrome/browser/ui/views/frame/browser_native_widget_mac.mm
@@ -269,6 +269,7 @@ void BrowserNativeWidgetMac::ValidateUse
+    case IDC_DUPLICATE_TAB:
     case IDC_PIN_TARGET_TAB:
     case IDC_PRINT:
     case IDC_RELOAD:
--- a/chrome/app/chrome_dll.rc
+++ b/chrome/app/chrome_dll.rc
@@ -84,6 +84,8 @@ BEGIN
     VK_F5,          IDC_RELOAD_BYPASSING_CACHE, VIRTKEY, CONTROL
     VK_F5,          IDC_RELOAD_BYPASSING_CACHE, VIRTKEY, SHIFT
     "T",            IDC_RESTORE_TAB,            VIRTKEY, CONTROL, SHIFT
+    "D",            IDC_DUPLICATE_TAB,          VIRTKEY, CONTROL
+    "Z",            IDC_RESTORE_TAB,            VIRTKEY, CONTROL
     "S",            IDC_CTRL_S_SHORTCUT,        VIRTKEY, CONTROL
     "9",            IDC_SELECT_LAST_TAB,        VIRTKEY, CONTROL
     VK_NUMPAD9,     IDC_SELECT_LAST_TAB,        VIRTKEY, CONTROL
