--- a/chrome/browser/ui/views/frame/horizontal_tab_strip_region_view.cc
+++ b/chrome/browser/ui/views/frame/horizontal_tab_strip_region_view.cc
@@ -542,6 +542,10 @@ void HorizontalTabStripRegionView::Initi
 }
 
 void HorizontalTabStripRegionView::ResetTabStrip() {
+  if (!tab_strip_set_) {
+    return;
+  }
+
   tab_strip_set_ = false;
   static_cast<BrowserTabStripController*>(tab_strip_->controller())->Reset();
   tab_strip_->Reset();
