--- a/chrome/browser/ui/views/tabs/tab_icon.cc
+++ b/chrome/browser/ui/views/tabs/tab_icon.cc
@@ -13,17 +13,12 @@
 #include "chrome/browser/browser_process.h"
 #include "chrome/browser/favicon/favicon_utils.h"
 #include "chrome/browser/themes/theme_properties.h"
-#include "chrome/browser/ui/browser.h"
 #include "chrome/browser/ui/browser_element_identifiers.h"
-#include "chrome/browser/ui/browser_finder.h"
-#include "chrome/browser/ui/browser_window.h"
 #include "chrome/browser/ui/color/chrome_color_id.h"
 #include "chrome/browser/ui/layout_constants.h"
 #include "chrome/browser/ui/tabs/tab_renderer_data.h"
-#include "chrome/browser/ui/user_education/browser_user_education_interface.h"
 #include "chrome/browser/ui/views/dotted_icon.h"
 #include "chrome/common/webui_url_constants.h"
-#include "components/feature_engagement/public/feature_constants.h"
 #include "components/grit/components_scaled_resources.h"
 #include "components/performance_manager/public/user_tuning/prefs.h"
 #include "components/prefs/pref_service.h"
@@ -478,12 +473,6 @@ void TabIcon::SetDiscarded(bool discarde
           gfx::Animation::RichAnimationDuration(base::Seconds(1)));
       tab_discard_animation_.Start();
       favicon_size_animation_.Hide();
-
-      // Potentially show an IPH if a tab was discarded.
-      Browser* browser = chrome::FindBrowserWithUiElementContext(
-          views::ElementTrackerViews::GetInstance()->GetContextForView(this));
-      BrowserUserEducationInterface::From(browser)->MaybeShowFeaturePromo(
-          feature_engagement::kIPHDiscardRingFeature);
     } else {
       tab_discard_animation_.Stop();
       favicon_size_animation_.Show();
