--- a/chrome/browser/ui/views/frame/layout/browser_view_layout.h
+++ b/chrome/browser/ui/views/frame/layout/browser_view_layout.h
@@ -102,7 +102,7 @@ class BrowserViewLayout : public views::
   // very small window, even on large monitors (which is why a minimum height is
   // not specified). This value is used for the main browser window only, not
   // for popups.
-  static constexpr int kMainBrowserContentsMinimumWidth = 500;
+  static constexpr int kMainBrowserContentsMinimumWidth = 400;
 
   // The minimum width of the contents area itself. Applies even when side
   // panels are open and prevents zero or negative contents sizes.
--- a/chrome/browser/ui/views/frame/layout/browser_view_tabbed_layout_impl.cc
+++ b/chrome/browser/ui/views/frame/layout/browser_view_tabbed_layout_impl.cc
@@ -811,18 +811,16 @@ gfx::Size BrowserViewTabbedLayoutImpl::G
                    std::max({side_panel_size.height(), main_area_size.height(),
                              vertical_tabstrip_size.height()});
 
-  // This assumes a horizontal tabstrip. There is also a hard minimum on the
-  // width of the browser defined by `kMainBrowserContentsMinimumWidth`.
-  int min_width = vertical_tabstrip_size.width() +
-                  std::max({horizontal_tabstrip_size.width(),
-                            side_panel_size.width() + main_area_size.width(),
-                            kMainBrowserContentsMinimumWidth});
+  // The browser window should have the same minimum width in all layouts.
+  // Controls & stacked side views (VTS + split view + side panel) may overflow
+  // or overlap at this width, but that's inevitable at scale this small.
+  // If the user prefers this width, that's their choice and responsibility.
+  const int min_width = kMainBrowserContentsMinimumWidth;
 
   // Maybe adjust for additional padding when the side panel is visible.
   if (side_panel_size.width() > 0 && !delegate().IsActiveTabSplit()) {
     const auto padding = GetLayoutConstant(LayoutConstant::kSidePanelInset);
     min_height += 2 * padding;
-    min_width += padding;
   }
 
   return gfx::Size(min_width, min_height);
