--- a/chrome/browser/ui/views/tabs/vertical/vertical_tab_strip_view.cc
+++ b/chrome/browser/ui/views/tabs/vertical/vertical_tab_strip_view.cc
@@ -154,10 +154,6 @@ views::ProposedLayout VerticalTabStripVi
   const int unpinned_preferred_height =
       unpinned_tabs_scroll_view_->GetPreferredSize(size_bounds).height();
 
-  const bool should_show_separator = pinned_preferred_height != 0 &&
-                                     unpinned_preferred_height != 0 &&
-                                     is_collapsed_;
-
   // If the height is bounded, calculate the available space for laying out the
   // pinned and unpinned containers.
   int remaining_height = 0;
@@ -166,10 +162,6 @@ views::ProposedLayout VerticalTabStripVi
     if (pinned_preferred_height != 0 && unpinned_preferred_height != 0) {
       remaining_height -= region_vertical_padding;
     }
-    if (should_show_separator) {
-      remaining_height -= tabs_separator_->GetPreferredSize().height() +
-                          region_vertical_padding;
-    }
     // Clamp the remaining height to 0 if we have less space.
     remaining_height = std::max(remaining_height, 0);
   }
@@ -203,29 +195,6 @@ views::ProposedLayout VerticalTabStripVi
     }
   }
 
-  // Place the tabs separator if visible.
-  if (should_show_separator) {
-    int separator_width =
-        size_bounds.width().value() - 2 * region_horizontal_padding;
-    int separator_x = region_horizontal_padding;
-    if (is_collapsed_) {
-      const int collapsed_separator_width = GetLayoutConstant(
-          LayoutConstant::kVerticalTabStripCollapsedSeparatorWidth);
-      separator_width = collapsed_separator_width;
-      separator_x = (size_bounds.width().value() - separator_width) / 2;
-    }
-    gfx::Rect tabs_separator_bounds(
-        separator_x, y, separator_width,
-        tabs_separator_->GetPreferredSize().height());
-    layouts.child_layouts.emplace_back(tabs_separator_.get(), true,
-                                       tabs_separator_bounds);
-
-    y += tabs_separator_bounds.height() + region_vertical_padding;
-  } else {
-    layouts.child_layouts.emplace_back(tabs_separator_.get(), false,
-                                       gfx::Rect());
-  }
-
   // Place the unpinned container using the entire available width, we do not
   // inset the x value by |region_horizontal_padding| here because, when the tab
   // strip is collapsed, tab groups need to draw the group colored line in this
