Index: src/components/kaguya_services/pref_names.h
===================================================================
--- src.orig/components/kaguya_services/pref_names.h	2026-06-20 10:13:04.415227680 +0800
+++ src/components/kaguya_services/pref_names.h	2026-06-20 10:17:05.345500348 +0800
@@ -25,6 +25,9 @@
 inline constexpr char kKaguyaBangsEnabled[] =
     "kaguya.services.bangs";
 
+inline constexpr char kKaguyaUpdateFetchingEnabled[] =
+    "kaguya.services.browser_updates";
+
 inline constexpr char kKaguyaSpellcheckEnabled[] =
     "kaguya.services.spellcheck_files";
 
Index: src/components/kaguya_services/kaguya_services_helpers.h
===================================================================
--- src.orig/components/kaguya_services/kaguya_services_helpers.h	2026-06-20 10:13:04.415166771 +0800
+++ src/components/kaguya_services/kaguya_services_helpers.h	2026-06-20 10:17:05.345672640 +0800
@@ -22,11 +22,13 @@
 
 COMPONENT_EXPORT(KAGUYA) bool ShouldFetchBangs(const PrefService& prefs);
 COMPONENT_EXPORT(KAGUYA) bool ShouldAccessExtensionService(const PrefService& prefs);
+COMPONENT_EXPORT(KAGUYA) bool ShouldAccessUpdateService(const PrefService& prefs);
 COMPONENT_EXPORT(KAGUYA) GURL GetServicesBaseURL(const PrefService& prefs);
 COMPONENT_EXPORT(KAGUYA) GURL GetDummyURL();
 COMPONENT_EXPORT(KAGUYA) GURL GetExtensionUpdateURL(const PrefService& prefs);
 COMPONENT_EXPORT(KAGUYA) GURL GetWebstoreSnippetURL(const PrefService& prefs, std::string_view id);
 COMPONENT_EXPORT(KAGUYA) GURL GetSpellcheckURL(const PrefService& prefs);
+COMPONENT_EXPORT(KAGUYA) GURL GetBrowserUpdateURL(const PrefService& prefs);
 COMPONENT_EXPORT(KAGUYA) std::optional<GURL> GetValidUserOverridenURL(std::string_view user_url_);
 COMPONENT_EXPORT(KAGUYA) void ConfigurePrefChangeRegistrarFor(std::string_view pref_name,
                             PrefChangeRegistrar& registrar, const base::RepeatingClosure& observer);
Index: src/components/kaguya_services/kaguya_services_helpers.cc
===================================================================
--- src.orig/components/kaguya_services/kaguya_services_helpers.cc	2026-06-20 10:13:04.415054284 +0800
+++ src/components/kaguya_services/kaguya_services_helpers.cc	2026-06-20 10:17:05.345758011 +0800
@@ -16,6 +16,14 @@
 
 namespace kaguya {
 
+#if BUILDFLAG(IS_MAC)
+#if defined(ARCH_CPU_ARM_FAMILY)
+constexpr std::string cpu_arch = "arm64";
+#else
+constexpr std::string cpu_arch = "x86_64";
+#endif
+#endif
+
 std::optional<GURL> GetValidUserOverridenURL(std::string_view user_url_) {
     if (user_url_.empty()) {
         return std::nullopt;
@@ -72,6 +80,11 @@
             prefs.GetBoolean(prefs::kKaguyaBangsEnabled);
 }
 
+bool ShouldAccessUpdateService(const PrefService& prefs) {
+    return ShouldAccessServices(prefs) &&
+            prefs.GetBoolean(prefs::kKaguyaUpdateFetchingEnabled);
+}
+
 GURL GetExtensionUpdateURL(const PrefService& prefs) {
     if (!ShouldAccessExtensionService(prefs)) {
         return GetDummyURL();
@@ -97,6 +110,20 @@
     return GetServicesBaseURL(prefs).Resolve("/dict/");
 }
 
+GURL GetBrowserUpdateURL(const PrefService& prefs) {
+    if (!ShouldAccessUpdateService(prefs)) {
+        return GetDummyURL();
+    }
+
+#if BUILDFLAG(IS_MAC)
+    std::string path = base::StringPrintf(
+        "/updates/mac/appcast-%s.xml", cpu_arch);
+    return GetServicesBaseURL(prefs).Resolve(path);
+#else
+    return GetDummyURL();
+#endif
+}
+
 void ConfigurePrefChangeRegistrarFor(std::string_view pref_name,
     PrefChangeRegistrar& registrar, const base::RepeatingClosure& observer) {
   registrar.Add(prefs::kKaguyaServicesEnabled, observer);
Index: src/chrome/browser/resources/settings/privacy_page/services_page.html
===================================================================
--- src.orig/chrome/browser/resources/settings/privacy_page/services_page.html	2026-06-20 10:13:04.416714864 +0800
+++ src/chrome/browser/resources/settings/privacy_page/services_page.html	2026-06-20 10:17:05.345851542 +0800
@@ -72,6 +72,11 @@
                     sub-label="$i18n{kaguyaSpellcheckToggleDescription}">
                 </settings-toggle-button>
             </if>
+            <settings-toggle-button id="updatesToggleButton"
+                pref="{{prefs.kaguya.services.browser_updates}}"
+                label="$i18n{kaguyaUpdatesToggle}"
+                sub-label="$i18n{kaguyaUpdatesToggleDescription}">
+            </settings-toggle-button>
         </cr-collapse>
     </div>
 </div>
Index: src/chrome/browser/ui/webui/settings/settings_localized_strings_provider.cc
===================================================================
--- src.orig/chrome/browser/ui/webui/settings/settings_localized_strings_provider.cc	2026-06-20 10:13:04.417912906 +0800
+++ src/chrome/browser/ui/webui/settings/settings_localized_strings_provider.cc	2026-06-20 10:17:05.346262056 +0800
@@ -2251,6 +2251,10 @@
       {"kaguyaSpellcheckToggle", IDS_SETTINGS_KAGUYA_SERVICES_SPELLCHECK_TOGGLE},
       {"kaguyaSpellcheckToggleDescription",
         IDS_SETTINGS_KAGUYA_SERVICES_SPELLCHECK_TOGGLE_DESCRIPTION},
+      {"kaguyaUpdatesToggle",
+        IDS_SETTINGS_KAGUYA_SERVICES_UPDATE},
+      {"kaguyaUpdatesToggleDescription",
+        IDS_SETTINGS_KAGUYA_SERVICES_UPDATE_DESCRIPTION},
       {"kaguyaOriginOverride", IDS_SETTINGS_KAGUYA_SERVICES_OVERRIDE},
       {"kaguyaOriginOverrideDescription",
         IDS_SETTINGS_KAGUYA_SERVICES_OVERRIDE_DESCRIPTION},
Index: src/chrome/browser/extensions/api/settings_private/prefs_util.cc
===================================================================
--- src.orig/chrome/browser/extensions/api/settings_private/prefs_util.cc	2026-06-20 10:13:04.416322364 +0800
+++ src/chrome/browser/extensions/api/settings_private/prefs_util.cc	2026-06-20 10:17:05.346781441 +0800
@@ -403,6 +403,8 @@
       settings_api::PrefType::kBoolean;
   (*s_allowlist)[::prefs::kKaguyaSpellcheckEnabled] =
       settings_api::PrefType::kBoolean;
+  (*s_allowlist)[::prefs::kKaguyaUpdateFetchingEnabled] =
+      settings_api::PrefType::kBoolean;
   (*s_allowlist)[::prefs::kKaguyaServicesOrigin] =
       settings_api::PrefType::kString;
 
Index: src/chrome/browser/ui/browser_ui_prefs.cc
===================================================================
--- src.orig/chrome/browser/ui/browser_ui_prefs.cc	2026-06-20 10:13:04.416568838 +0800
+++ src/chrome/browser/ui/browser_ui_prefs.cc	2026-06-20 10:17:05.347035534 +0800
@@ -226,6 +226,7 @@
     registry->RegisterBooleanPref(prefs::kKaguyaBangsEnabled, true);
     registry->RegisterBooleanPref(prefs::kKaguyaExtProxyEnabled, true);
     registry->RegisterBooleanPref(prefs::kKaguyaSpellcheckEnabled, true);
+    registry->RegisterBooleanPref(prefs::kKaguyaUpdateFetchingEnabled, true);
     registry->RegisterStringPref(prefs::kKaguyaServicesOrigin, "");
   }
 
Index: src/chrome/app/settings_strings.grdp
===================================================================
--- src.orig/chrome/app/settings_strings.grdp	2026-06-20 10:13:05.613779960 +0800
+++ src/chrome/app/settings_strings.grdp	2026-06-20 10:17:05.347466738 +0800
@@ -2008,6 +2008,19 @@
   <message name="IDS_SETTINGS_KAGUYA_SERVICES_SPELLCHECK_TOGGLE_DESCRIPTION" desc="Description of the for enabling/disabling of downloading spellcheck files">
     Kaguya will fetch dictionary files used for spell checking when requested. When disabled, spell checking will not work.
   </message>
+  <message name="IDS_SETTINGS_KAGUYA_SERVICES_UPDATE" desc="Toggle for automatic update downloads">
+    Allow automatic browser and component updates
+  </message>
+  <if expr="is_macosx">
+    <message name="IDS_SETTINGS_KAGUYA_SERVICES_UPDATE_DESCRIPTION" desc="Description of the toggle for automatic update downloads">
+      Kaguya will automatically download and install browser and component updates as they become available. We recommend keeping this setting enabled to ensure you get the latest features and security updates.
+    </message>
+  </if>
+  <if expr="not is_macosx">
+    <message name="IDS_SETTINGS_KAGUYA_SERVICES_UPDATE_DESCRIPTION" desc="Description of the toggle for automatic update downloads">
+      Kaguya will automatically download and install browser and component updates as they become available. We recommend keeping this setting enabled to ensure you get the latest features and security updates. Automatic core browser updates are not available on this platform yet, but component updates are. Please use external software to keep Kaguya up to date.
+    </message>
+  </if>
   <message name="IDS_SETTINGS_KAGUYA_SERVICES_OVERRIDE" desc="Text input for overriding the Kaguya services server">
     Use your own instance of Kaguya services
   </message>
