--- a/chrome/browser/ui/toasts/toast_controller.cc
+++ b/chrome/browser/ui/toasts/toast_controller.cc
@@ -19,6 +19,8 @@
 #include "base/time/time.h"
 #include "base/timer/timer.h"
 #include "chrome/browser/browser_process.h"
+#include "chrome/browser/ui/browser_element_identifiers.h"
+#include "chrome/browser/ui/tabs/features.h"
 #include "chrome/browser/ui/browser_window/public/browser_window_features.h"
 #include "chrome/browser/ui/browser_window/public/browser_window_interface.h"
 #include "chrome/browser/ui/browser_window/public/desktop_browser_window_capabilities.h"
@@ -31,12 +33,14 @@
 #include "chrome/browser/ui/toasts/toast_metrics.h"
 #include "chrome/browser/ui/toasts/toast_view.h"
 #include "chrome/browser/ui/webui_browser/webui_browser.h"
+#include "chrome/browser/ui/views/interaction/browser_elements_views.h"
 #include "chrome/common/pref_names.h"
 #include "components/omnibox/common/omnibox_focus_state.h"
 #include "components/prefs/pref_service.h"
 #include "components/tabs/public/tab_interface.h"
 #include "content/public/browser/web_contents.h"
 #include "ui/base/l10n/l10n_util.h"
+#include "ui/base/interaction/element_identifier.h"
 #include "ui/views/bubble/bubble_dialog_delegate_view.h"
 #include "ui/views/focus/focus_manager.h"
 #include "ui/views/view.h"
@@ -46,6 +50,7 @@
 #include "chrome/browser/ui/fullscreen_util_mac.h"
 #endif
 
+
 ToastParams::ToastParams(ToastId id) : toast_id(id) {}
 ToastParams::ToastParams(ToastParams&& other) noexcept = default;
 ToastParams& ToastParams::operator=(ToastParams&& other) noexcept = default;
@@ -292,7 +297,12 @@ void ToastController::CreateToast(ToastP
     return;
   }
 
-  views::View* const anchor_view = browser_window_interface_->TopContainer();
+  views::View* anchor_view = browser_window_interface_->TopContainer();
+  if (tabs::AreHeliumVerticalTabsEnabled()) {
+    anchor_view = BrowserElementsViews::From(browser_window_interface_)
+      ->GetView(kToolbarAppMenuButtonElementId)
+      ->parent(); // ToolbarView::container_view_
+  }
   CHECK(anchor_view);
   const ui::ImageModel* image_override = params.image_override.has_value()
                                              ? &params.image_override.value()
