--- a/chrome/browser/ui/views/extensions/extension_installed_bubble_view.cc
+++ b/chrome/browser/ui/views/extensions/extension_installed_bubble_view.cc
@@ -124,24 +124,6 @@ void ShowExtensionPostInstallDialog(
       .SetIcon(ui::ImageModel::FromImageSkia(
           weak_delegate->model()->MakeIconOfSize(kMaxIconSize)));
 
-  if (weak_delegate->model()->show_how_to_use()) {
-    dialog_model_builder.AddParagraph(
-        ui::DialogModelLabel(weak_delegate->model()->GetHowToUseText()));
-  }
-  if (weak_delegate->model()->show_key_binding()) {
-    dialog_model_builder.AddParagraph(
-        ui::DialogModelLabel::CreateWithReplacement(
-            IDS_EXTENSION_INSTALLED_MANAGE_SHORTCUTS,
-            ui::DialogModelLabel::CreateLink(
-                IDS_EXTENSION_INSTALLED_MANAGE_SHORTCUTS_LINK_TEXT,
-                base::BindRepeating(&ExtensionInstalledBubbleView::LinkClicked,
-                                    base::Unretained(weak_delegate)))));
-  }
-
-  if (weak_delegate->model()->show_how_to_manage()) {
-    dialog_model_builder.AddParagraph(ui::DialogModelLabel(
-        l10n_util::GetStringUTF16(IDS_EXTENSION_INSTALLED_MANAGE_INFO)));
-  }
 #if !BUILDFLAG(IS_CHROMEOS)
   // Add a sync or sign in promo in the footer if it should be shown.
   extensions::ExtensionRegistry* registry =
