--- a/ui/native_theme/features/native_theme_features.cc
+++ b/ui/native_theme/features/native_theme_features.cc
@@ -19,7 +19,13 @@ constexpr base::FeatureState kOverlayScr
 // Enables or disables overlay scrollbars in Blink (i.e. web content) on Aura
 // or Linux.  The status of native UI overlay scrollbars is determined in
 // PlatformStyle::CreateScrollBar. Does nothing on Mac.
-BASE_FEATURE(kOverlayScrollbar, kOverlayScrollbarFeatureState);
+BASE_FEATURE(kOverlayScrollbar,
+#if BUILDFLAG(IS_WIN) || BUILDFLAG(IS_LINUX)
+  base::FEATURE_ENABLED_BY_DEFAULT
+#else
+  kOverlayScrollbarFeatureState
+#endif
+);
 
 // Disable to keep scrollbars visible forever once shown, and immediately
 // update scrollbar states instead of animating. This is used to ensure
