--- a/chrome/browser/ui/views/frame/browser_frame_view_mac.mm
+++ b/chrome/browser/ui/views/frame/browser_frame_view_mac.mm
@@ -186,7 +186,7 @@ gfx::Rect BrowserFrameViewMac::GetBounds
     // The bottom curve of the first/last tab swoops into the caption button
     // region, so account for this when calculating insets.
     const gfx::Insets insets = GetCaptionButtonInsets(
-        /*visual_overlap=*/TabStyle::Get()->GetBottomCornerRadius());
+        /*visual_overlap=*/TabStyle::Get()->GetBottomCornerRadius() / 2);
     bounds.Inset(insets);
   }
 
--- a/chrome/browser/ui/views/frame/browser_frame_view_win.cc
+++ b/chrome/browser/ui/views/frame/browser_frame_view_win.cc
@@ -234,10 +234,13 @@ bool BrowserFrameViewWin::CaptionButtons
 
 gfx::Rect BrowserFrameViewWin::GetBoundsForTabStripRegion(
     const gfx::Size& tabstrip_minimum_size) const {
-  const int x = CaptionButtonsOnLeadingEdge() ? CaptionButtonsRegionWidth() : 0;
+  const int x = CaptionButtonsOnLeadingEdge() ?
+      CaptionButtonsRegionWidth() :
+      GetLayoutConstant(TOOLBAR_CORNER_RADIUS);
   int end_x = width();
   if (!CaptionButtonsOnLeadingEdge()) {
-    end_x = std::min(width() - CaptionButtonsRegionWidth(), end_x);
+    end_x = std::min(width() - CaptionButtonsRegionWidth() -
+            GetLayoutConstant(TOOLBAR_CORNER_RADIUS), end_x);
   }
   return gfx::Rect(x, TopAreaHeight(false), std::max(0, end_x - x),
                    tabstrip_minimum_size.height());
