--- a/chrome/browser/ui/views/frame/layout/browser_view_layout_delegate_impl.cc
+++ b/chrome/browser/ui/views/frame/layout/browser_view_layout_delegate_impl.cc
@@ -258,7 +258,7 @@ BrowserViewLayoutDelegateImpl::GetBounds
                    layout.trailing_exclusion.vertical_padding);
   bounds_f.set_height(
       std::max(max_bound, static_cast<float>(tabstrip_minimum_size.height())));
-  const int tab_margin = TabStyle::Get()->GetBottomCornerRadius();
+  const int tab_margin = TabStyle::Get()->GetBottomCornerRadius() / 2;
   bounds_f.Inset(gfx::InsetsF::TLBR(
       0.0f,
       layout.leading_exclusion.content.width() +
--- a/chrome/browser/ui/views/frame/browser_frame_view_win.cc
+++ b/chrome/browser/ui/views/frame/browser_frame_view_win.cc
@@ -258,10 +258,13 @@ bool BrowserFrameViewWin::CaptionButtons
 
 gfx::Rect BrowserFrameViewWin::GetBoundsForTabStripRegion(
     const gfx::Size& tabstrip_minimum_size) const {
-  const int x = CaptionButtonsOnLeadingEdge() ? CaptionButtonsRegionWidth() : 0;
+  const int x = CaptionButtonsOnLeadingEdge() ?
+      CaptionButtonsRegionWidth() :
+      GetLayoutConstant(TOOLBAR_CORNER_RADIUS);
   int end_x = width();
   if (!CaptionButtonsOnLeadingEdge()) {
-    end_x = std::min(width() - CaptionButtonsRegionWidth(), end_x);
+    end_x = std::min(width() - CaptionButtonsRegionWidth() -
+            GetLayoutConstant(TOOLBAR_CORNER_RADIUS), end_x);
   }
   return gfx::Rect(x, TopAreaHeight(false), std::max(0, end_x - x),
                    tabstrip_minimum_size.height());
