--- a/components/neterror/resources/dino_game/game_over_panel.ts
+++ b/components/neterror/resources/dino_game/game_over_panel.ts
@@ -12,7 +12,7 @@ import type {SpritePosition} from './spr
 import type {Trex} from './trex.js';
 import {getTimeStamp} from './utils.js';
 
-const RESTART_ANIM_DURATION: number = 875;
+const RESTART_ANIM_DURATION: number = 640;
 const LOGO_PAUSE_DURATION: number = 875;
 const FLASH_ITERATIONS: number = 5;
 
@@ -210,6 +210,10 @@ export class GameOverPanel {
       restartSourceHeight *= 2;
       framePosX *= 2;
     }
+    // Clear previous pixels.
+    this.canvasCtx.clearRect(
+      restartTargetX, restartTargetY, dimensions.restartWidth,
+      dimensions.restartHeight);
 
     this.canvasCtx.save();
 
@@ -262,8 +266,12 @@ export class GameOverPanel {
     } else if (
         this.currentFrame > 0 && this.currentFrame < animConfig.frames.length) {
       if (this.animTimer >= animConfig.msPerFrame) {
+        this.animTimer = 0;
         this.currentFrame++;
-        this.drawRestartButton();
+
+        if (this.currentFrame < animConfig.frames.length) {
+          this.drawRestartButton();
+        }
       }
     } else if (
         !this.altGameModeActive &&
