--- a/components/password_manager/core/browser/password_suggestion_generator.cc
+++ b/components/password_manager/core/browser/password_suggestion_generator.cc
@@ -400,6 +400,8 @@ PasswordSuggestionGenerator::PasswordSug
 
 void PasswordSuggestionGenerator::AppendOptionalFooterSection(
     std::vector<autofill::Suggestion>* suggestions) const {
+  return;
+
   bool has_webauthn_credential = std::ranges::any_of(
       *suggestions,
       [](SuggestionType type) {
