--- a/chrome/browser/ui/views/toolbar/toolbar_view.cc
+++ b/chrome/browser/ui/views/toolbar/toolbar_view.cc
@@ -268,18 +268,6 @@ void ToolbarView::Init() {
   aura::WindowOcclusionTracker::ScopedPause pause_occlusion;
 #endif
 
-  // The background views must be behind container_view_.
-  leading_curve_ = AddChildViewAt(std::make_unique<View>(), 0);
-  leading_curve_->SetBackground(
-      std::make_unique<TabStripLikeBackground>(browser_view_));
-  trailing_curve_ = AddChildViewAt(std::make_unique<View>(), 0);
-  trailing_curve_->SetBackground(
-      std::make_unique<TabStripLikeBackground>(browser_view_));
-
-  active_state_subscription_ =
-      GetWidget()->RegisterPaintAsActiveChangedCallback(base::BindRepeating(
-          &ToolbarView::ActiveStateChanged, base::Unretained(this)));
-
   auto location_bar = std::make_unique<LocationBarView>(
       browser_, browser_->profile(), browser_->command_controller(), this,
       display_mode_ != DisplayMode::kNormal);
@@ -815,19 +803,6 @@ void ToolbarView::Layout(PassKey) {
   // The container view should be the exact same size/position as ToolbarView.
   container_view_->SetSize(size());
 
-  // The background views should be behind the top-left and top-right corners
-  // of the container_view_.
-  const int corner_radius =
-      GetLayoutConstant(LayoutConstant::kToolbarCornerRadius);
-  const auto [leading_corner_style, trailing_corner_style] = GetCornerStyles();
-  const int leading_curve_size =
-      leading_corner_style == CornerStyle::kTabstripCurve ? corner_radius : 0;
-  const int trailing_curve_size =
-      trailing_corner_style == CornerStyle::kTabstripCurve ? corner_radius : 0;
-  leading_curve_->SetBounds(0, 0, leading_curve_size, leading_curve_size);
-  trailing_curve_->SetBounds(width() - trailing_curve_size, 0,
-                             trailing_curve_size, trailing_curve_size);
-
   if (display_mode_ == DisplayMode::kCustomTab) {
     custom_tab_bar_->SetBounds(0, 0, width(),
                                custom_tab_bar_->GetPreferredSize().height());
@@ -837,9 +812,6 @@ void ToolbarView::Layout(PassKey) {
 
   if (display_mode_ == DisplayMode::kNormal) {
     LayoutCommon();
-    UpdateClipPath(
-        leading_corner_style != CornerStyle::kSquare ? corner_radius : 0,
-        trailing_corner_style != CornerStyle::kSquare ? corner_radius : 0);
   }
 
   if (toolbar_controller_) {
@@ -877,41 +849,6 @@ void ToolbarView::OnThemeChanged() {
   SchedulePaint();
 }
 
-void ToolbarView::UpdateClipPath(int leading_corner_radius,
-                                 int trailing_corner_radius) {
-  const gfx::Rect local_bounds = GetLocalBounds();
-  // The bottom of the toolbar may be clipped more than necessary in
-  // certain scale factor so adds extra 2dp so that even if the origin
-  // and the height are rounded down, we still can paint til the
-  // bottom of the toolbar. The similar logic is applied to
-  // BookmarkBarView which can be the bottom component within the
-  // TopContainerView, and TopContainerView which is the parent and
-  // can also clip the paint region for child views.
-  // TODO(crbug.com/41344902): Remove this hack once the pixel canvas is
-  // enabled on all aura platforms.
-  const int extended_height = local_bounds.height() + 2;
-  const SkPath path =
-      SkPathBuilder()
-          .moveTo(0, local_bounds.height())
-          .lineTo(0, leading_corner_radius)
-          .arcTo(SkVector(leading_corner_radius, leading_corner_radius), 0,
-                 SkPathBuilder::kSmall_ArcSize, SkPathDirection::kCW,
-                 SkPoint(leading_corner_radius, 0))
-          .lineTo(local_bounds.width() - trailing_corner_radius, 0)
-          .arcTo(SkVector(trailing_corner_radius, trailing_corner_radius), 0,
-                 SkPathBuilder::kSmall_ArcSize, SkPathDirection::kCW,
-                 SkPoint(local_bounds.width(), trailing_corner_radius))
-          .lineTo(local_bounds.width(), extended_height)
-          .lineTo(0, extended_height)
-          .detach();
-  container_view_->SetClipPath(path);
-}
-
-void ToolbarView::ActiveStateChanged() {
-  leading_curve_->SchedulePaint();
-  trailing_curve_->SchedulePaint();
-}
-
 void ToolbarView::NewTabButtonPressed(const ui::Event& event) {
   chrome::NewTab(browser_view_->browser(),
                  NewTabTypes::kNewTabButtonInToolbarForTouch);
