--- a/chrome/browser/ui/views/frame/layout/browser_view_layout_delegate_impl.cc
+++ b/chrome/browser/ui/views/frame/layout/browser_view_layout_delegate_impl.cc
@@ -127,12 +127,7 @@ bool BrowserViewLayoutDelegateImpl::IsIn
 }
 
 bool BrowserViewLayoutDelegateImpl::IsContentsSeparatorEnabled() const {
-  // Web app windows manage their own separator.
-  // TODO(crbug.com/40102629): Make PWAs set the visibility of the ToolbarView
-  // based on whether it is visible instead of setting the height to 0px. This
-  // will enable BrowserViewLayout to hide the contents separator on its own
-  // using the same logic used by normal BrowserElementsViews.
-  return !browser_view_->browser()->app_controller();
+  return false;
 }
 
 bool BrowserViewLayoutDelegateImpl::IsActiveTabSplit() const {
--- a/chrome/browser/ui/views/frame/multi_contents_view.cc
+++ b/chrome/browser/ui/views/frame/multi_contents_view.cc
@@ -659,38 +659,11 @@ void MultiContentsView::OnDragAndDropPre
   InvalidateLayout();
 }
 
-void MultiContentsView::SetShouldShowTopSeparator(bool should_show) {
-  if (contents_separators_.should_show_top == should_show) {
-    return;
-  }
-  contents_separators_.should_show_top = should_show;
+void MultiContentsView::SetShouldShowTopSeparator(bool should_show) {}
 
-  // This can be called during BrowserView layout, so protect against creating a
-  // layout loop.
-  InvalidateLayout(/*avoid_propagate_during_layout=*/true);
-}
-
-void MultiContentsView::SetShouldShowLeadingSeparator(bool should_show) {
-  if (contents_separators_.should_show_leading == should_show) {
-    return;
-  }
-  contents_separators_.should_show_leading = should_show;
-
-  // This can be called during BrowserView layout, so protect against creating a
-  // layout loop.
-  InvalidateLayout(/*avoid_propagate_during_layout=*/true);
-}
-
-void MultiContentsView::SetShouldShowTrailingSeparator(bool should_show) {
-  if (contents_separators_.should_show_trailing == should_show) {
-    return;
-  }
-  contents_separators_.should_show_trailing = should_show;
+void MultiContentsView::SetShouldShowLeadingSeparator(bool should_show) {}
 
-  // This can be called during BrowserView layout, so protect against creating a
-  // layout loop.
-  InvalidateLayout(/*avoid_propagate_during_layout=*/true);
-}
+void MultiContentsView::SetShouldShowTrailingSeparator(bool should_show) {}
 
 BEGIN_METADATA(MultiContentsView)
 END_METADATA
