--- a/chrome/browser/ui/views/side_panel/side_panel.cc
+++ b/chrome/browser/ui/views/side_panel/side_panel.cc
@@ -444,7 +444,7 @@ bool SidePanel::IsRightAligned() const {
 gfx::Size SidePanel::GetMinimumSize() const {
   const int min_height = 0;
   return gfx::Size(
-      SidePanelEntry::kSidePanelDefaultContentWidth + GetBorderInsets().width(),
+      SidePanelEntry::kSidePanelMinimumContentWidth + GetBorderInsets().width(),
       min_height);
 }
 
--- a/chrome/browser/ui/views/side_panel/side_panel_entry.h
+++ b/chrome/browser/ui/views/side_panel/side_panel_entry.h
@@ -38,8 +38,10 @@ class SidePanelEntry final : public ui::
     kToolbar,
   };
 
-  // The default and minimum acceptable side panel content width.
+  // The default side panel content width.
   static constexpr int kSidePanelDefaultContentWidth = 360;
+  // The minimum acceptable side panel content width
+  static constexpr int kSidePanelMinimumContentWidth = 260;
   using CreateContentCallback =
       base::RepeatingCallback<std::unique_ptr<views::View>(
           SidePanelEntryScope&)>;
