--- a/chrome/browser/ui/views/side_panel/side_panel_resize_area.cc
+++ b/chrome/browser/ui/views/side_panel/side_panel_resize_area.cc
@@ -29,7 +29,7 @@ SidePanelResizeHandle::SidePanelResizeHa
     : side_panel_(side_panel) {
   SetProperty(views::kElementIdentifierKey, kSidePanelResizeHandleElementId);
   SetVisible(false);
-  gfx::Size preferred_size(4, 24);
+  gfx::Size preferred_size(2, 24);
   SetPreferredSize(preferred_size);
   SetCanProcessEventsWithinSubtree(false);
 
