--- a/chrome/browser/ui/views/side_panel/side_panel_entry.h
+++ b/chrome/browser/ui/views/side_panel/side_panel_entry.h
@@ -39,7 +39,7 @@ class SidePanelEntry final : public ui::
   };
 
   // The default and minimum acceptable side panel content width.
-  static constexpr int kSidePanelDefaultContentWidth = 360;
+  static constexpr int kSidePanelDefaultContentWidth = 260;
   using CreateContentCallback =
       base::RepeatingCallback<std::unique_ptr<views::View>(
           SidePanelEntryScope&)>;
