--- a/chrome/browser/ui/views/frame/browser_view.cc
+++ b/chrome/browser/ui/views/frame/browser_view.cc
@@ -4916,6 +4916,18 @@ int BrowserView::NonClientHitTest(const
     }
   }
 
+  // The empty spaces in toolbar should allow mouse events to fall
+  // through to the caption area.
+  if (toolbar_) {
+    gfx::Point test_point(point);
+    if (ConvertedHitTest(parent(), toolbar_, &test_point)) {
+      if (toolbar_->IsPositionInWindowCaption(test_point)) {
+        return HTCAPTION;
+      }
+      return HTCLIENT;
+    }
+  }
+
   // Let the frame handle any events that fall within the bounds of the window
   // controls overlay.
   if (IsWindowControlsOverlayEnabled() && GetActiveWebContents()) {
--- a/chrome/browser/ui/views/toolbar/toolbar_view.cc
+++ b/chrome/browser/ui/views/toolbar/toolbar_view.cc
@@ -183,86 +183,6 @@ struct CaptionHitTestData {
   int after_dist = 0;
 };
 
-// Calculates the `CaptionHitTestData` (`data`) starting at `view` for `point`,
-// which should be in `view`'s local coordinates. Bails out immediately if a
-// View is hit; may traverse into icon containers.
-void CalculateIsPositionInWindowCaption(CaptionHitTestData& data,
-                                        const views::View* view,
-                                        const gfx::Point& point) {
-  for (auto& child : view->children()) {
-    if (!child->GetVisible()) {
-      continue;
-    }
-    const gfx::Rect bounds = child->bounds();
-
-    if (views::IsViewClass<ToolbarIconContainerView>(child) ||
-        views::IsViewClass<page_actions::PageActionContainerView>(child)) {
-      // Traverse into known icon containers.
-      const auto in_child =
-          views::View::ConvertPointToTarget(view, child, point);
-      CalculateIsPositionInWindowCaption(data, child, in_child);
-    } else if (bounds.x() <= point.x() && bounds.right() >= point.x()) {
-      // This point is in/above/below the child.
-      data.at = child;
-      data.is_direct_hit = bounds.Contains(point);
-    } else {
-      // See if the view is the closest before or after the target point in the
-      // layout.
-      if (bounds.right() < point.x()) {
-        const int dist = point.x() - bounds.right();
-        if (!data.before || data.before_dist > dist) {
-          data.before = child;
-          data.before_dist = dist;
-        }
-      } else if (bounds.x() > point.x()) {
-        const int dist = bounds.x() - point.x();
-        if (!data.after || data.after_dist > dist) {
-          data.after = child;
-          data.after_dist = dist;
-        }
-      }
-    }
-
-    // If a view was hit at any level, stop processing.
-    if (data.at) {
-      break;
-    }
-  }
-}
-
-// Returns whether `point` should be treated as part of the caption area in
-// `view`, which should be the topmost view in the browser.
-bool IsPositionInWindowCaption(const views::View* view,
-                               const gfx::Point& point) {
-  CaptionHitTestData data;
-  CalculateIsPositionInWindowCaption(data, view, point);
-
-  const bool is_above_centerline =
-      point.y() <= view->GetLocalBounds().CenterPoint().y();
-  const auto is_separator = [](const views::View* view) {
-    return views::IsViewClass<views::Separator>(view) ||
-           views::IsViewClass<ToolbarDivider>(view);
-  };
-
-  // If the point is in a view, then it's not in the caption unless the view is
-  // a separator. If the point is at a view but not in it, then it is caption if
-  // the point is centerline; otherwise it's not.
-  if (data.at) {
-    return is_separator(data.at) ||
-           (!data.is_direct_hit && is_above_centerline);
-  }
-
-  // If the point is not in a view but it is next to a separator or the edge of
-  // the toolbar, it is caption.
-  if (!data.before || is_separator(data.before) || !data.after ||
-      is_separator(data.after)) {
-    return true;
-  }
-
-  // All remaining points (between non-separator views) are caption if they are
-  // above centerline.
-  return is_above_centerline;
-}
 // Gets the display mode for a given browser.
 ToolbarView::DisplayMode GetDisplayMode(Browser* browser) {
   // Checked in this order because even tabbed PWAs use the CUSTOM_TAB
@@ -1227,6 +1147,44 @@ bool ToolbarView::GetAppMenuFocused() co
   return app_menu_button_ && app_menu_button_->HasFocus();
 }
 
+bool ToolbarView::IsRectInWindowCaption(const gfx::Rect& rect) const {
+  // Clickable elements are child views of the toolbar. If the toolbar is not
+  // visible, then the rect is in the window caption.
+  if (!GetVisible()) {
+    return true;
+  }
+
+  const auto get_target_rect = [&](views::View* target) {
+    gfx::RectF rect_in_target_coords_f(rect);
+    views::View::ConvertRectToTarget(this, target, &rect_in_target_coords_f);
+    return gfx::ToEnclosingRect(rect_in_target_coords_f);
+  };
+
+  // Check each child view to see if the rect intersects with
+  // any clickable elements. If it does, check if the click is actually on that
+  // element. False if on a clickable element, true if not on a clickable element.
+  for (views::View* const child : children()) {
+    if (!child->GetVisible()) {
+      continue;
+    }
+
+    const gfx::Rect target_rect = get_target_rect(child);
+    if (child->GetLocalBounds().Intersects(target_rect)) {
+      // Check if the rect is actually on the child.
+      if (child->HitTestRect(target_rect)) {
+        return false;
+      }
+    }
+  }
+
+  // The rect is not on any clickable element, so it's in the window caption.
+  return true;
+}
+
+bool ToolbarView::IsPositionInWindowCaption(const gfx::Point& point) const {
+  return IsRectInWindowCaption(gfx::Rect(point, gfx::Size(1, 1)));
+}
+
 void ToolbarView::ShowIntentPickerBubble(
     std::vector<IntentPickerBubbleView::AppInfo> app_info,
     bool show_stay_in_chrome,
@@ -1263,11 +1221,6 @@ void ToolbarView::ShowBookmarkBubble(con
                                  already_bookmarked);
 }
 
-bool ToolbarView::IsPositionInWindowCaption(
-    const gfx::Point& test_point) const {
-  return ::IsPositionInWindowCaption(this, test_point);
-}
-
 views::Button* ToolbarView::GetChromeLabsButton() const {
   return browser_->GetFeatures()
       .chrome_labs_coordinator()
--- a/chrome/browser/ui/views/toolbar/toolbar_view.h
+++ b/chrome/browser/ui/views/toolbar/toolbar_view.h
@@ -146,6 +146,15 @@ class ToolbarView : public views::Access
   // Returns true if the app menu is focused.
   bool GetAppMenuFocused() const;
 
+  // Returns true if the specified rect is not on a clickable element,
+  // aka container view children: toolbar buttons, location bar, etc.
+  bool IsRectInWindowCaption(const gfx::Rect& rect) const;
+
+  // A convenience function which calls |IsRectInWindowCaption()| with a rect of
+  // size 1x1 and an origin of |point|. |point| is in the local coordinate space
+  // of |this|.
+  bool IsPositionInWindowCaption(const gfx::Point& point) const;
+
   void ShowIntentPickerBubble(
       std::vector<IntentPickerBubbleView::AppInfo> app_info,
       bool show_stay_in_chrome,
@@ -157,12 +166,6 @@ class ToolbarView : public views::Access
   // Shows a bookmark bubble and anchors it appropriately.
   void ShowBookmarkBubble(const GURL& url, bool already_bookmarked);
 
-  // Used to test whether `test_point` should be treated as part of the caption
-  // bar, which means it can be used to drag the window or open the window
-  // context menu. Should only be called when the toolbar is in the caption
-  // area.
-  bool IsPositionInWindowCaption(const gfx::Point& test_point) const;
-
   // Accessors.
   Browser* browser() const { return browser_; }
   views::Button* GetChromeLabsButton() const;
