--- a/chrome/browser/ui/views/frame/themed_background.cc
+++ b/chrome/browser/ui/views/frame/themed_background.cc
@@ -32,7 +32,6 @@ std::pair<int, ui::ColorId> GetThemeChoi
     const views::View* view) {
   switch (theme_choice) {
     case ThemedBackground::ThemeChoice::kToolbarTheme:
-      return {IDR_THEME_TOOLBAR, kColorToolbar};
     case ThemedBackground::ThemeChoice::kFrameTheme:
       if (IsFrameActive(view)) {
         return {IDR_THEME_FRAME, ui::kColorFrameActive};
