--- a/chrome/browser/ui/views/frame/top_container_background.cc
+++ b/chrome/browser/ui/views/frame/top_container_background.cc
@@ -8,6 +8,7 @@
 
 #include "chrome/browser/themes/theme_properties.h"
 #include "chrome/browser/ui/color/chrome_color_id.h"
+#include "chrome/browser/ui/views/frame/browser_frame_view.h"
 #include "chrome/browser/ui/views/frame/browser_view.h"
 #include "chrome/browser/ui/views/tabs/tab_strip.h"
 #include "chrome/grit/theme_resources.h"
@@ -79,7 +80,8 @@ void TopContainerBackground::PaintBackgr
                                              const BrowserView* browser_view) {
   bool painted = PaintThemeCustomImage(canvas, view, browser_view);
   if (!painted) {
-    canvas->DrawColor(view->GetColorProvider()->GetColor(kColorToolbar));
+    canvas->DrawColor(browser_view->browser_widget()->GetFrameView()->GetFrameColor(
+        BrowserFrameActiveState::kUseCurrent));
   }
 }
 
@@ -88,7 +90,8 @@ std::optional<SkColor> TopContainerBackg
     const BrowserView* browser_view) {
   const bool will_be_painted = WillPaintCustomImage(view);
   if (!will_be_painted) {
-    return view->GetColorProvider()->GetColor(kColorToolbar);
+    return browser_view->browser_widget()->GetFrameView()->GetFrameColor(
+        BrowserFrameActiveState::kUseCurrent);
   }
 
   return std::nullopt;
--- a/chrome/browser/ui/views/frame/layout/browser_view_tabbed_layout_impl.cc
+++ b/chrome/browser/ui/views/frame/layout/browser_view_tabbed_layout_impl.cc
@@ -46,6 +46,8 @@ BrowserViewTabbedLayoutImpl::~BrowserVie
 
 BrowserViewTabbedLayoutImpl::TopSeparatorType
 BrowserViewTabbedLayoutImpl::GetTopSeparatorType() const {
+  return TopSeparatorType::kNone;
+
   if (!delegate().IsToolbarVisible() && !delegate().IsBookmarkBarVisible()) {
     return TopSeparatorType::kNone;
   }
