--- a/chrome/browser/ui/views/frame/layout/browser_view_layout_delegate_impl.cc
+++ b/chrome/browser/ui/views/frame/layout/browser_view_layout_delegate_impl.cc
@@ -127,6 +127,7 @@ bool BrowserViewLayoutDelegateImpl::IsIn
 }
 
 bool BrowserViewLayoutDelegateImpl::IsContentsSeparatorEnabled() const {
+  return false;
   // Web app windows manage their own separator.
   // TODO(crbug.com/40102629): Make PWAs set the visibility of the ToolbarView
   // based on whether it is visible instead of setting the height to 0px. This
--- a/chrome/browser/ui/views/frame/top_container_background.cc
+++ b/chrome/browser/ui/views/frame/top_container_background.cc
@@ -8,6 +8,7 @@
 
 #include "chrome/browser/themes/theme_properties.h"
 #include "chrome/browser/ui/color/chrome_color_id.h"
+#include "chrome/browser/ui/views/frame/browser_frame_view.h"
 #include "chrome/browser/ui/views/frame/browser_view.h"
 #include "chrome/browser/ui/views/tabs/tab_strip.h"
 #include "chrome/grit/theme_resources.h"
@@ -32,7 +33,6 @@ std::pair<int, ui::ColorId> GetColorChoi
     const views::View* view) {
   switch (color_choice) {
     case TopContainerBackground::ColorChoice::kToolbarColor:
-      return {IDR_THEME_TOOLBAR, kColorToolbar};
     case TopContainerBackground::ColorChoice::kFrameColor:
       if (IsFrameActive(view)) {
         return {IDR_THEME_FRAME, ui::kColorFrameActive};
--- a/chrome/browser/ui/views/frame/layout/browser_view_tabbed_layout_impl.cc
+++ b/chrome/browser/ui/views/frame/layout/browser_view_tabbed_layout_impl.cc
@@ -51,6 +51,8 @@ BrowserViewTabbedLayoutImpl::~BrowserVie
 
 BrowserViewTabbedLayoutImpl::TopSeparatorType
 BrowserViewTabbedLayoutImpl::GetTopSeparatorType() const {
+  return TopSeparatorType::kNone;
+
   if (!delegate().IsToolbarVisible() && !delegate().IsBookmarkBarVisible()) {
     return TopSeparatorType::kNone;
   }
