--- a/chrome/browser/resources/settings/clear_browsing_data_dialog/clear_browsing_data_dialog.css
+++ b/chrome/browser/resources/settings/clear_browsing_data_dialog/clear_browsing_data_dialog.css
@@ -61,16 +61,6 @@
   position: fixed;
 }
 
-#manageOtherGoogleDataRow {
-  border-radius: var(--dbd-container-border-radius);
-  color: var(--cr-primary-text-color);
-  background: var(--dbd-container-color);
-  margin-top: 12px;
-  display: flex;
-  align-items: center;
-  font-weight: 500;
-}
-
 #showMoreButton {
   --cr-button-background-color: var(--dbd-container-color);
   --cr-button-height: auto;
--- a/chrome/browser/resources/settings/clear_browsing_data_dialog/clear_browsing_data_dialog.html.ts
+++ b/chrome/browser/resources/settings/clear_browsing_data_dialog/clear_browsing_data_dialog.html.ts
@@ -47,12 +47,6 @@ export function getHtml(this: SettingsCl
           role="presentation">
       </cr-icon>
     </cr-button>
-    <cr-link-row id="manageOtherGoogleDataRow"
-        label="${this.computeOtherGoogleDataRowLabel_()}"
-        sub-label="${this.computeOtherGoogleDataRowSubLabel_()}"
-        @click="${this.onManageOtherGoogleDataRowClick_}"
-        role-description="$i18n{subpageArrowRoleDescription}">
-    </cr-link-row>
   </div>
   <div slot="button-container" class="row-aligned">
 <if expr="not is_chromeos">
--- a/chrome/browser/resources/settings/clear_browsing_data_dialog/clear_browsing_data_dialog.ts
+++ b/chrome/browser/resources/settings/clear_browsing_data_dialog/clear_browsing_data_dialog.ts
@@ -9,7 +9,6 @@
 import 'chrome://resources/cr_elements/cr_button/cr_button.js';
 import 'chrome://resources/cr_elements/cr_dialog/cr_dialog.js';
 import 'chrome://resources/cr_elements/cr_icon/cr_icon.js';
-import 'chrome://resources/cr_elements/cr_link_row/cr_link_row.js';
 import '../controls/settings_checkbox.js';
 // <if expr="not is_chromeos">
 import './clear_browsing_data_account_indicator.js';
@@ -23,7 +22,6 @@ import {SyncBrowserProxyImpl} from '/sha
 import {PrefService} from '/shared/settings/prefs2/pref_service.js';
 import type {CrButtonElement} from 'chrome://resources/cr_elements/cr_button/cr_button.js';
 import type {CrDialogElement} from 'chrome://resources/cr_elements/cr_dialog/cr_dialog.js';
-import type {CrLinkRowElement} from 'chrome://resources/cr_elements/cr_link_row/cr_link_row.js';
 import {WebUiListenerMixinLit} from 'chrome://resources/cr_elements/web_ui_listener_mixin_lit.js';
 import {assert, assertNotReached, assertNotReachedCase} from 'chrome://resources/js/assert.js';
 import {FocusOutlineManager} from 'chrome://resources/js/focus_outline_manager.js';
@@ -68,7 +66,6 @@ export interface SettingsClearBrowsingDa
     deleteButton: CrButtonElement,
     deleteBrowsingDataDialog: CrDialogElement,
     deletingDataAlert: HTMLElement,
-    manageOtherGoogleDataRow: CrLinkRowElement,
     moreOptionsList: HTMLElement,
     showMoreButton: CrButtonElement,
     spinner: HTMLElement,
@@ -177,7 +174,7 @@ export class SettingsClearBrowsingDataDi
     };
   }
 
-  protected accessor dataTypesExpanded_: boolean = false;
+  protected accessor dataTypesExpanded_: boolean = true;
   protected accessor deletingDataAlertString_: string = '';
   protected accessor isDeletionInProgress_: boolean = false;
   private accessor isNoDatatypeSelected_: boolean = false;
@@ -488,8 +485,6 @@ export class SettingsClearBrowsingDataDi
   protected onOtherGoogleDataDialogCancel_(e: Event) {
     e.stopPropagation();
     this.showOtherGoogleDataDialog_ = false;
-    this.updateComplete.then(
-        () => focusWithoutInk(this.$.manageOtherGoogleDataRow));
   }
 
   protected onSubLabelLinkClicked_(e: CustomEvent<{id: string}>) {
