--- a/chrome/browser/resources/settings/languages_page/languages_page.html.ts
+++ b/chrome/browser/resources/settings/languages_page/languages_page.html.ts
@@ -30,9 +30,6 @@ export function getHtml(this: SettingsLa
           <div title="${item.language.nativeDisplayName}" role="none">
             ${this.formatIndex_(index)}. ${item.language.displayName}
           </div>
-          <div class="target-info secondary">
-            $i18n{translateTargetLabel}
-          </div>
 <if expr="is_win">
           <div class="explain-selected"
               ?hidden="${!this.isProspectiveUiLanguage_(
--- a/chrome/browser/resources/settings/languages_page/translate_page.html.ts
+++ b/chrome/browser/resources/settings/languages_page/translate_page.html.ts
@@ -8,6 +8,7 @@ import type {SettingsTranslatePageElemen
 
 export function getHtml(this: SettingsTranslatePageElement) {
   return html`<!--_html_template_start_-->
+<if expr="False">
 <settings-section page-title="$i18n{translatePageTitle}">
 <settings-toggle-button id="offerTranslateOtherLanguages"
     pref-key="translate.enabled"
@@ -107,5 +108,6 @@ ${this.showAddNeverTranslateDialog_ ? ht
   </settings-add-languages-dialog>
 ` : ''}
 </settings-section>
+</if>
 <!--_html_template_end_-->`;
 }
